src/test/java/edu/hm/hafner/analysis/ReportPrinterTest.java

```diff
@@ -52,6 +57,189 @@ void shouldPrintAllIssuesToPrintStream() {
+        report.add(new IssueBuilder().setSeverity(Severity.WARNING_HIGH).setMessage("Severity Warning High").build());
+        Logger logger = mock(LoggerFactory.getLogger(SLF4JAdapter.class).getClass());
+        report.print(new SLF4JAdapter(logger));
+
+        for(Issue issue : report) {
+            verify(logger).warn(issue.toString());
+            verify(logger, never()).error(issue.toString());
+            verify(logger, never()).info(issue.toString());
+            verify(logger, never()).trace(issue.toString());
+        }
+    }
+
+    @Test
+    void shouldLogOneWarningNormalSF4J() {
+        Report report = new Report();
+        report.add(new IssueBuilder().setSeverity(Severity.WARNING_NORMAL).setMessage("Severity Warning Normal").build());
+        Logger logger = mock(LoggerFactory.getLogger(SLF4JAdapter.class).getClass());
+        report.print(new SLF4JAdapter(logger));
+
+        for(Issue issue : report) {
+            verify(logger).info(issue.toString());
+            verify(logger, never()).error(issue.toString());
+            verify(logger, never()).trace(issue.toString());
+            verify(logger, never()).warn(issue.toString());
+        }
+    }
+
+    @Test
+    void shouldLogOneWarningLowSLF4J() {
+        Report report = new Report();
+        report.add(new IssueBuilder().setSeverity(Severity.WARNING_LOW).setMessage("Severity Warning Low").build());
+        Logger logger = mock(LoggerFactory.getLogger(SLF4JAdapter.class).getClass());
+        report.print(new SLF4JAdapter(logger));
+
+        for(Issue issue : report) {
+            verify(logger).trace(issue.toString());
+            verify(logger, never()).error(issue.toString());
+            verify(logger, never()).info(issue.toString());
+            verify(logger, never()).warn(issue.toString());
+        }
+    }
+
+    @Test
+    void shouldLogAllIssuesSLF4J() {
+        Report report = readCheckStyleReport();
+        Logger logger = mock(LoggerFactory.getLogger(SLF4JAdapter.class).getClass());
+        report.print(new SLF4JAdapter(logger));
+
+        verify(logger).error(anyString());
+        verify(logger).warn(anyString());
+        verify(logger).info(anyString());
+        verify(logger).trace(anyString());
+    }
+
+    @Test
+    void shouldLogTwoIssuesSLF4J() {
+        Report report = new Report();
+        Issue s1 = new IssueBuilder().setSeverity(Severity.WARNING_LOW).setMessage("Severity Warning Low").build();
+        Issue s2 = new IssueBuilder().setSeverity(Severity.ERROR).setMessage("Severity Error").build();
+        report.add(s1);
+        report.add(s2);
+        Logger logger = mock(LoggerFactory.getLogger(SLF4JAdapter.class).getClass());
+        report.print(new SLF4JAdapter(logger));
+
+        verify(logger).error(s2.toString());
+        verify(logger).trace(s1.toString());
+        verify(logger, never()).warn(anyString());
+        verify(logger, never()).info(anyString());
+    }
```

Comment:
Evtl. könnte man auch alle einzelnen Tests zusammen fassen und in einer Methode abhandeln...